### ISSUE TITLE
An error occurred while pushing the image httpd-24 to Developer Sandbox cluster - Error: (HTTP code 500) server error - [object Object]

### ISSUE THREAD

**Issue by odockal** [redhat-developer/podman-desktop-sandbox-ext#122]
If I try to push some of the images I have locally (some from registry.redhat.io, some other from publicly accessible sources like registry.access.redhat.com), there is an error from dev sandbox.

```
An error occurred while pushing the image 'registry.access.redhat.com/ubi8/httpd-24:latest' to Developer Sandbox cluster 'dev-sandbox-context'. Error: (HTTP code 500) server error - [object Object]
```

Some image can be pushed:
```
docker.io/library/httpd:latest
quay.io/odockal/alpine-local
docker.io/library/alpine
```

Others cannot:
```
registry.redhat.io/rhel8/nginx-122:latest
registry.redhat.io/ubi8/httpd-24:latest
registry.access.redhat.com/ubi8/httpd-24:latest
```

See https://github.com/apocas/docker-modem/issues/171 issue about `[object Object]` in error message.

**Comment by dgolovin** [comment]
Most likely yhe image is signed and you have to add option to push command to remove signature. The criptic error is on docker-modem side, they don't capture the error property of the exception. I have issue opened there with pull request.

Try to push from cli and you will see actual error.

**Comment by dgolovin** [comment]
```
podman push default-route-openshift-image-registry.apps.sandbox-m3.1530.p1.openshiftapps.com/nivologd-dev/httpd-24
Getting image source signatures
Checking if image destination supports signatures
Error: Copying this image would require changing layer representation, which we cannot do: "Would invalidate signatures"
```

**Comment by odockal** [comment]
@dgolovin Ok, so we need to let user know and also offer a way how to do it?

**Comment by dgolovin** [comment]
This message is coming from docker-modem see (https://github.com/apocas/docker-modem/issues/171).  Until it is fixed, we cannot even detect the error and let user know what to do.

**Comment by odockal** [comment]
@dgolovin ah, alright. thanks for explanation, let's keep the issue opened then until the fix is at place.

**Comment by odockal** [comment]
@dgolovin So I tried to override the signatures locally, to have a workaround, but I am missing an authentication: 
```
podman push default-route-openshift-image-registry.apps.sandbox-m3.1530.p1.openshiftapps.com/odockal-dev/httpd-24 --remove-signatures
Error: trying to reuse blob sha256:ba181bba861a60bb8f268a6efe89e6421b78b3cfb8113585e2a4ebab6bc57f63 at destination: checking whether a blob sha256:ba181bba861a60bb8f268a6efe89e6421b78b3cfb8113585e2a4ebab6bc57f63 exists in default-route-openshift-image-registry.apps.sandbox-m3.1530.p1.openshiftapps.com/odockal-dev/httpd-24: authentication required

```

**Comment by dgolovin** [comment]
yes, to do it manually, you have to add registry default-route-openshift-image-registry.apps.sandbox-m3.1530.p1.openshiftapps.com/odockal-dev/httpd-24 using any user name and the same token used in sandbox connection.

**Comment by dgolovin** [comment]
@odockal another option is to use 'podman push' instead of dockerode when https://github.com/containers/podman-desktop/issues/5990 is fixed. Detect the error and request user confirmation to use --remove-signatures option when pushing.

**Comment by dgolovin** [comment]
After fix in docker-modem was merged and podman-desktop picked it up with dockerode dependency update the error is much better now. I think we can close this issue. @odockal WDYT?

![Image](https://github.com/user-attachments/assets/5c48c708-3f59-4c09-ab9d-25a88384eb59)

**Comment by odockal** [comment]
@dgolovin Error looks much better now! Thx
